mlrun/api/db/sqldb/models/models_mysql.py

```diff
@@ -162,6 +164,47 @@ class Artifact(Base, mlrun.utils.db.HasStruct):
         def get_identifier_string(self) -> str:
             return f"{self.project}/{self.key}/{self.uid}"
 
+    class ArtifactV2(Base, mlrun.utils.db.BaseModel):
+        __tablename__ = "artifacts_v2"
+        __table_args__ = (
+            UniqueConstraint("uid", "project", "key", name="_artifacts_v2_uc"),
+        )
+
+        Label = make_label(__tablename__)
+        Tag = make_tag_v2(__tablename__)
+
+        id = Column(Integer, primary_key=True)
+        key = Column(String(255, collation=SQLCollationUtil.collation()))
+        project = Column(String(255, collation=SQLCollationUtil.collation()))
+        kind = Column(String(255, collation=SQLCollationUtil.collation()))
+        producer_id = Column(String(255, collation=SQLCollationUtil.collation()))
+        iteration = Column(Integer)
+        uid = Column(String(255, collation=SQLCollationUtil.collation()))
+        created = Column(
+            sqlalchemy.dialects.mysql.TIMESTAMP(fsp=3),
+            default=datetime.now(timezone.utc),
+        )
+        updated = Column(
+            sqlalchemy.dialects.mysql.TIMESTAMP(fsp=3),
+            default=datetime.now(timezone.utc),
+        )
+        _full_object = Column("object", sqlalchemy.dialects.mysql.MEDIUMBLOB)
```

Comment:
Also regarding what we discussed offline, the confluence is defining the full object to be of type JSON, while in the description you refer to the `updated` and `created`. But there is another consideration which is regarding the possible content that the user can put inside the artifact body, such as `dataframes`, `datetime` etc.. which would be hard to serialize. 

As I see it there could be 2 solutions
- separating the artifact struct from the artifact body, where the `struct` is stored as Json and the body is stored as blob. This will require quering and upstarting from/to the db to have the separation and merging capablities to return a full artifact ( We already implement that with tags and labels where we pop them from the artifact struct before we store it in the blob and then on querying we merge them back ) This will allow further on querying on the table even if we didn't have columns for them.
- Store it all as blob, but then we will need to query the whole blob to filter over the artifact ( less suggested )

Reply:
Discussed offline with @theSaarco , lets keep it as blob (Option 2), as we can have not serializable fields in the artifact status as well ( preview, stats etc.. ) and we don't want to go into this rabbit hole for now.